security-policies/dev/common.py

```diff
@@ -197,9 +321,16 @@ def format_json_in_text(text):
         return text
 
 
-def fix_code_blocks(text: str):
+def fix_code_blocks(text: str, rule_number: str, benchmark_id: str):
     text = add_new_line_after_period(text)
-    text = format_json_in_text(text)
+    if rule_number in {"1.17", "3.5"} and benchmark_id == "cis_aws":
+        text = format_json_in_text(text)
+    elif (rule_number in {"3.10", "3.11"} and benchmark_id == "cis_aws") or (
+        rule_number in {"5.1.5"} and benchmark_id == "cis_azure"
+    ):
+        text = format_json_in_string_command(text)
+    else:
+        text = format_and_fix_json_in_text(text)
     return check_and_fix_numbered_list(text)
```

Comment:
I’d prefer that we don’t apply a function tailored to a specific rule. Instead, we should implement a fallback mechanism, if the output doesn’t match the expected format or an error was thrown - we trigger the next JSON formatting function in line.